src/java/voldemort/utils/pool/KeyedResourcePool.java

```diff
@@ -557,9 +557,10 @@ private void throwReportedExceptions() throws Exception {
                 skippedExceptionCount++;
                 if(elapsedTime <= excpetionReportTimeMS) {
                     Exception e = entry.getSecond();
-                    logger.info(" Throwing remembered exception. time elapsed (ms) " + elapsedTime
-                                + ". Exception : "
-                                + e.getMessage());
+                    if(logger.isDebugEnabled()) {
```

Comment:
Though it could be used at many places, there is one mainline use and others are there, deprecated or for testing purposes.